hw/ip/ascon/data/ascon.hjson

```diff
@@ -0,0 +1,696 @@
+            or wait for a separate trigger signal before starting (1) (see Trigger Register). Default: 0
+          '''
+        }
+        { bits: "1",
+          name: "FORCE_DATA_OVERWRITE",
+          desc:  '''
+            Control whether the Ascon unit is stalled during the last
+            encryption/decryption cycle if the previous output data has not yet been read (0)
+            or finishes the operation and overwrites the previous output data (1). Default: 0
+          '''
+        }
+      ]
+    },
+    { name: "CTRL_AUX_REGWEN",
+      desc: '''
+        Lock bit for Auxiliary Control Register.
+        '''
+      swaccess: "rw0c",
+      hwaccess: "none",
+      fields: [
+      {   bits: "0",
+          desc: '''
+            Auxiliary Control Register configuration enable bit.
+            If this is cleared to 0, the Auxiliary Control Register cannot be written anymore.
+            ''',
+          resval: 1,
+        }
+      ]
+    },
+    { name: "BLOCK_CTRL_SHADOWED",
+      desc: '''
+        Block Control Register.
+        This register is shadowed, meaning two subsequent write operations are required to change its content.
+        If the two write operations try to set a different value, a recoverable alert is triggered (See Status Register).
+        A read operation clears the internal phase tracking:
+        The next write operation is always considered a first write operation of an update sequence.
+        This register has to be written at least for each first and last block of message, associated data.
+        Intermediate blocks are expected to be of full block size.
+        If there's only one block of data, all three fields: valid_bytes, data_type_last and data_type_start must be set.
+      '''
+      swaccess: "rw",
+      hwaccess: "hro",
+      hwext:    "false"
+      hwqe:     "true"
+      shadowed: "true"
+      fields: [
+        { bits: "11:0",
+          name: "DATA_TYPE_START",
+          desc:  '''
+            Specifies which input the ascon unit shall process next. There are:
+            PT_IN, CT_IN, AD_IN. They are internally mubi4 encoded. Only one type is allowed to be true at the same time.
+            NONE sets all mubi4
+          '''
+        }
+        { bits: "23:12",
+          name: "DATA_TYPE_LAST",
+          desc:  '''
+            Specifies that the next read from an input is the last of its kind. There are:
+            PT_IN, CT_IN, AD_IN. They are internally mubi4 encoded. Only one type is allowed to be true at the same time.
+            NONE sets all mubi4 values to false.
+          '''
+        }
+        { bits: "28:24",
+          name: "VALID_BYTES",
+          resval: "31",
+          desc:  '''
+            Indicates the number of valid bytes [0 to 16] bytes.
+            Default: 16 bytes for ASCON 128a
+            Default:   8 bytes for ASCON 128
```

Comment:
I think we should take a decision on what mode to implement or otherwise implement both but specify a mode field in the main control register to switch.

Reply:
Creating another issue to discuss and take a decision sounds good to me. According to a footnote in your draft spec, Ascon-128 is definitely needed (maybe both modes). So we should NOT include a synthesis switch IMO.